source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -186,6 +186,16 @@ SanitizerInterceptor::~SanitizerInterceptor() {
     DestroyShadowMemoryOnCPU();
     DestroyShadowMemoryOnPVC();
     DestroyShadowMemoryOnDG2();
+
+    m_Quarantine = nullptr;
+    m_MemBufferMap.clear();
```

Comment:
Please add comments to explain why we need to do this before releasing adapter.

Reply:
Done